WcaOnRails/app/models/competition.rb

```diff
@@ -427,16 +439,24 @@ def dangerously_close_to?(c)
     self.kilometers_to(c) <= NEARBY_DISTANCE_KM_DANGER && days_until.abs < NEARBY_DAYS_DANGER
   end
 
+  def results_uploaded?
+    Result.exists?(competitionId: self.id)
+  end
+
   def user_can_view?(user)
     self.showAtAll || (user && user.can_manage_competition?(self))
   end
 
+  def in_progress?
+    start_date <= Date.today && end_date >= Date.today
```

Comment:
Alternatives: `(start_date..end_date).cover? Date.today` and `(start_date..end_date) === Date.today`

btw. Calling `Date.today` twice might be dangerous if the code is executed right before midnight. The odds of this happening are rather low, though. ;)